torchdiffeq/_impl/misc.py

```diff
@@ -282,3 +302,26 @@ def np_nextafter(x1, x2):
     x2_np = x2.detach().cpu().numpy()
     out = torch.tensor(np.nextafter(x1_np, x2_np)).to(x1)
     return out.detach() + (x1 - x1.detach())  # stitch gradients.
+
+
+def _check_timelike(name, timelike, can_grad):
+    assert isinstance(timelike, torch.Tensor), '{} must be a torch.Tensor'.format(name)
+    _assert_floating(name, timelike)
+    assert timelike.ndimension() == 1, "{} must be one dimensional".format(name)
+    if not can_grad:
+        assert not timelike.requires_grad, "{} cannot require gradient".format(name)
+    diff = timelike[1:] > timelike[:-1]
+    assert diff.all() or (~diff).all(), '{} must be strictly increasing or decreasing'.format(name)
+    return timelike
```

Comment:
`timelike` isn't modified so doesn't need to be returned. (An error I see is present in my original PR.)

Reply:
Done.